ext/standard/string.c

```diff
@@ -4790,11 +4790,29 @@ PHP_FUNCTION(strip_tags)
 		Z_PARAM_ZVAL(allow)
 	ZEND_PARSE_PARAMETERS_END();
 
-	/* To maintain a certain BC, we allow anything for the second parameter and return original string */
 	if (allow) {
-		convert_to_string(allow);
-		allowed_tags = Z_STRVAL_P(allow);
-		allowed_tags_len = Z_STRLEN_P(allow);
+		if (Z_TYPE_P(allow) == IS_ARRAY) {
+			smart_str tags_ss = {0};
+			zval *tmp;
+
+			smart_str_alloc(&tags_ss, 0, 0);
+			ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(allow), tmp) {
+				convert_to_string_ex(tmp);
```

Comment:
To avoid modifying the array in-place, this should be something like `zend_string *tag = zval_get_string(tmp); ... zend_string_release(tag);`